routes/api/contacts.js

```diff
@@ -1,25 +1,78 @@
-const express = require('express')
+const contactsMethots = require("./contactsMethots.js");
+const express = require("express");
 
-const router = express.Router()
+const router = express.Router();
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
+  const contacts = await contactsMethots.listContacts();
+  res.status(200).json(contacts);
+});
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:id", async (req, res, next) => {
+  const id = req.params.id;
+  const contactsId = await contactsMethots.getContactById(id);
+  console.log(id);
+  res.status(200).json(contactsId);
+});
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post("/", async (req, res, next) => {
+  const { name, email, phone } = req.body;
```

Comment:
Zróbmy tutaj walidację, pominiesz wtedy wyżucanie i łapanie błędów z funkcji + wtedy też sprawdzi ci czy są odpowiednie pola oraz czy nie ma dodatkowych

```
router.post("/", async (req, res, next) => {
	try {
		const validation = schema.validate(req.body);
		if (validation.error) {
			res.status(400).json({ error: validation.error.message });
		}
		const contactsAdded = await contactsMethots.addContact(req.body);
		res.status(201).json(contactsAdded);
	} catch (error) {
		console.error("Error adding contact:", error);
		res.status(500).json({ error: "Internal server error" });
	}
});
```

i wtedy addContacts powinno wyglądać tak

```
async function addContact({ name, email, phone }) {
	const data = await fs.readFile(contactsPath, "utf-8");
	const contacts = JSON.parse(data);
	const lastContact = contacts[contacts.length - 1];
	const newId = parseInt(lastContact.id) + 1;
	const newContact = { id: newId.toString(), name, email, phone };
	contacts.push(newContact);
	await fs.writeFile(contactsPath, JSON.stringify(contacts));
	return newContact;
}
```
Tylko wtedy musisz przenieść scheme tutaj, albo dać ją do innego pliku i zaimportować. Zrób to samo dla reszty endpointów bo raczej to rzucanie błędów jest średnim pomysłem